src/Controller.java

```diff
@@ -149,27 +216,24 @@ public void actionPerformed(ActionEvent e)  {
                 clearFields();
             }
             // clearing fields
-            
-            
-            
-            
+                     
             //When a row is added, one of the food groups is marked as "eaten"
-            if (selectedCheckbox == "Fruits and Vegetables") {
+            if (selectedCheckbox.equals("Fruits and Vegetables")) {

```

Comment:
good practise: avoid hard-coded strings and put them in constant and use that constant